src/components/crossForm/CrossForm.vue

```diff
@@ -400,64 +533,137 @@ export default {
     this.isMounted = true
   },
   methods: {
+    async handleChangeNetwork(network) {
+      try {
+        const chainId = numToHex(network.networkId)
+        await window.ethereum.request({
+          method: 'wallet_switchEthereumChain',
+          params: [{ chainId }],
+        });
+        const networksConf = getNetworksConf(network.networkId)
+        this.originNetworkSelected = network
+        this.destinationNetworks = networksConf?.networks.map(record => record.crossToNetwork) 
+      } catch (error) {
+        if (error.code === 4902) {
+          await this.handleAddNetwork(this.sharedState.currentConfig.crossToNetwork)
+        }
+      }
+    },
+    handleChangeDestinationNetwork(network) {
+      const originNetwork = findNetworkByChainId(
+        this.originNetworkSelected.networkId,
+        network.networkId,
+      )
+      const rskConfig = originNetwork.isRsk ? originNetwork : originNetwork.crossToNetwork
+      const sideConfig = rskConfig.crossToNetwork
+      if (this.originNetworkSelected.crossToNetwork.networkId !== network.networkId) {
+        const originNetworkIndex = this.originNetworks.findIndex(
+          net => net.networkId === originNetwork.networkId,
+        )
+        this.originNetworks.splice(originNetworkIndex, 1, originNetwork)
+      }
+
+      store.initMainSettings(originNetwork.networkId, rskConfig, sideConfig)
+    },
+    async handleAddNetwork(networkConfig) {
+      try {
+        const chainId = numToHex(networkConfig.networkId)
+        await window.ethereum.request({
+          method: 'wallet_addEthereumChain',
+          params: [
+            {
+              chainId,
+              chainName: networkConfig.name,
+              nativeCurrency: {
+                name: networkConfig.mainToken.name,
+                symbol: networkConfig.mainToken.symbol,
+                decimals: networkConfig.mainToken.decimals,
+              },
+              rpcUrls: [networkConfig.rpc],
+            },
+          ],
+        })
+      } catch (error) {
+        console.error(error)
+      }
+    },
+    async handleSwitchNetwork() {
```

Comment:
This method looks too similar to handleChangeNetwork, maybe we should use only one of them

Reply:
it's similar, yet not the same. One is for switching origin-destination, and the other is to change the origin to something else.